### PR TITLE
HDF5/ASDF Comparison

### PR DESCRIPTION
**Pull request by perrygreenfield**
The edited version used for PLATO. If someone know how to deal with the odd breaks in the displayed code, let me know. I used rst2pdf to generate the PDFs.

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
Can you copy in the `.pre-commit-config.yaml` from `asdf`? That way pre-commit is happy. It might fix-up some things..

**Comment by WilliamJamieson** [comment]
> I used rst2pdf to generate the PDFs.

Also, once this is in I could go ahead and setup RTD to generate the docs for this if you want.

File: general/hdf5_asdf_comparison.rst [hunk 1]
```diff
@@ -0,0 +1,6501 @@
...
+* `Appendix C: ASDF GWCS Contents`_
+* `Appendix D: Edited version of Appendix C`_
+* `Appendix E: Portion of converted HDF5 file`_
+
+.. _Introduction:
+
+Introduction
+============
+
+Based on a number of questions and comments about why we aren't using HDF5, that the first ASDF paper apparently hasn't addressed suitably well, this document is an attempt to explain in more detail why it hasn't been chosen for STScI use as a data format.
```
Comment (nden) [inline general/hdf5_asdf_comparison.rst:29]: Should we have a reference or a link to the ASDF paper?

File: general/hdf5_asdf_comparison.rst [hunk 2]
```diff
@@ -0,0 +1,6501 @@
...
+
+For completeness, the original list of concerns (condensed) were:
+
+
+1. Entirely binary.
+2. Not self documenting
+3. Effectively only one implementation (due to complexity)
+4. Questionable as an archival format
+5. Does not lend itself to pure text-based data files as an option
+6. HDF5 Abstract Data Model not flexible enough.
```
Comment (nden) [inline general/hdf5_asdf_comparison.rst]:
```suggestion
6. HDF5 Abstract Data Model not flexible enough

```

File: general/hdf5_asdf_comparison.rst [hunk 3]
```diff
@@ -0,0 +1,6501 @@
...
+
+
+There are two uses of the YAML “anchor” and “alias” features in this ASDF example. This feature allows associating an “anchor”, essentially a special identifier, to a node in the tree that other parts of the tree can refer to through the use of “aliases” that references that special identifier. In this way another node may point to the contents of a different node for reference, avoiding having to copy the contents and risk the two nodes becoming different in the future. In this example one of the referenced nodes was fairly large (but small compared to the total tree), but only had one alias that referenced it. There were several references to a much smaller node. All these aliases were replaced by copies in the ASDF to avoid the work of supporting such aliases in the conversion.
+
+
+A small segment of the contents of the HDF5 file is displayed in Appendix E to contrast the difficulty of directly interpreting the contents as compared to ASDF. In the ASCII rendition one can see where some keys are located, but not much beyond that.
```
Comment (nden) [inline general/hdf5_asdf_comparison.rst]:
```suggestion
A small segment of the contents of the HDF5 file is displayed in `Appendix E`_ to contrast the difficulty of directly interpreting the contents as compared to ASDF. In the ASCII rendition one can see where some keys are located, but not much beyond that.

```

Comment (nden) [reply general/hdf5_asdf_comparison.rst]: added a link

File: general/hdf5_asdf_comparison.rst (hunk 1 above)
Comment (perrygreenfield) [reply general/hdf5_asdf_comparison.rst:29]: I added one.

File: general/hdf5_asdf_comparison.rst (hunk 3 above)
Comment (perrygreenfield) [reply general/hdf5_asdf_comparison.rst]: fixed reference to Appendix E.

**Comment by perrygreenfield** [comment]
@WilliamJamieson I did add the pre-commit file. Sure, if you can set up RTD for this that would be great.

**Comment by WilliamJamieson** [comment]
> @WilliamJamieson I did add the pre-commit file. Sure, if you can set up RTD for this that would be great.

PR #2 adds the ability to support docs building. I will setup RTD for this repo shortly.

**Review by nden — APPROVED** [review]

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
Can `hdf5_asdf_comparison.rst` be moved into the `docs` directory?

After that can it be added to a `.. toctree::` inside the `index.rst` file?

**Pull request merged**